rasa_core/server.py

```diff
@@ -534,6 +551,38 @@ def create_app(agent=None,
                                 "Evaluation could not be created. Error: {}"
                                 "".format(e))
 
+    @app.post("/intentEvaluation")
+    @requires_auth(app, auth_token)
+    async def evaluate_intents(request: Request):
+        """Evaluate intents against a Rasa NLU model."""
+
+        tmpdir = tempfile.mkdtemp()
+
+        zipped_model_path = os.path.join(tmpdir, 'model.tar.gz')
+        with open(zipped_model_path, 'w+b') as f:
+            f.write(request.body)
+
+        unzipped_path = unpack_model(zipped_model_path, tmpdir)
```

Comment:
we should move this logic out of the endpoint into a separate function (so that we can test separately without running the server)